syllabus/1-Cryptography/zk-proofs/slides.md

```diff
@@ -0,0 +1,446 @@
+---
+title: ZK proofs
+description: Introduction to zero-knowledge proofs and zk-SNARKS
+duration: 3.5 hour
+---
+
+# ZK Proofs
+
+---
+
+# Outline
+
+<pba-flex center>
+
+1. [ZK Proofs overview](#zk-proofs)
+2. [Examples](#simple-zk-example)
+3. [zk-SNARKs](#zk-snarks)
+4. [Applications of zk-SNARKs](#zk-application)
+5. [Under the hood of zk-SNARKs](#zk-practice)
+6. [Generating ZK-proofs using Circom and snarkjs](#circom-snarkjs)
+
+</pba-flex>
+
+---
+
+## ZK Proofs
+
+How do we do private operations on a public blockchain<br />and have everyone know that they were done correctly?
+
+Notes:
+
+(we are working on substrate support for these and will use them for protocols)
+
+---
+
+## What is a ZK Proof?
+
+- A prover wants to convince a verifier that something is true without revealing why it is true.
+
+- They are mostly interactive protocols, but mostly we'll turn them  non-interactive and deal with their non-interactive variety. 
+---
+
+## What can we show?
+
+- NP relation: `function(statement, witness) -> bool`
+
+- Prover knows a witness for a statement:
+
+  - They want to show that they know it (_a proof of knowledge_)
+
+  - ... Without revealing anything about the witness (_ZK_)
+
+---
+
+## ZK Proof Interface
+
+- NP relation: `function(statement, witness) -> bool`
+
+- `prove(statement, witness) -> proof`
+
+- `verify(statement, proof) -> bool`
+
+---
+
+## Examples
+---
+
+## ZK Proof Example
+
+_Example:_ Schnorr signatures are ZK Proofs
+- $(s, e)$ such that $s = (\textrm{Random blinding factor }r) - \\ \textrm{Private Key} \times e$
+- They show that the prover knows the private key (the discrete log of the public key) without revealing anything about it.
```

Comment:
I think you should tread lightly on this. Even mentioning discrete log I dont even think is necessary. I think it can just be generalized to the prover contains knowledge of a secret key and can prove it to the verifier without revealing it.

Something to this affect but if you do decide to continue just remember to tread lightly.